coveo/ui-kit
# fix(test): fix false positives from remove component assertion

https://coveord.atlassian.net/browse/KIT-2497

I originally thought this was a problem with passing a selector, but it's actually more likely that there is a problem with Cypress itself for the `.should('not.exist')` case. Although no bug has been reported for this on Cypress' side (at least not that I've found), I've found a few threads where others are experiencing a very similar problem where `not.exist` isn't behaving as it should. After lots of research/tinkering (too much, honestly), I've decided to instead just check for a specific console error that is thrown whenever a component is placed outside of the result template and removed from the DOM. It's not ideal, but it's better than what was there before 😅 

To test:
1. Add a `assertRemovesComponent` call to a test where we know a component _**should**_ exist and ensure it fails
2. Add a `assertRemovesComponent` call to a test where we know a component _**won't**_ exist and ensure it passes
